src/pages/Collections/CollectionPage.ViewModel.ts

```diff
@@ -0,0 +1,325 @@
+   * @param collectionId
+   * @param folderId
+   * @param uuid
+   * @returns
+   */
+  private readRequestInFolder = (
+    collectionId: string,
+    folderId: string,
+    uuid: string,
+  ) => {
+    return this.collectionRepository.readRequestInFolder(
+      collectionId,
+      folderId,
+      uuid,
+    );
+  };
+
+  /**
+   * Retrieve request or folder from repository
+   * @param collectionId
+   * @param uuid
+   * @returns
+   */
+  private readRequestOrFolderInCollection = (
+    collectionId: string,
+    uuid: string,
+  ): Promise<CollectionItem> => {
+    return this.collectionRepository.readRequestOrFolderInCollection(
+      collectionId,
+      uuid,
+    );
+  };
+
+  /**
+   * Retrieve collection from repository
+   * @param uuid
+   * @returns
+   */
+  private readCollection = (uuid: string): Promise<CollectionDocument> => {
+    return this.collectionRepository.readCollection(uuid);
+  };
+
+  /**
+   * Save API Request from unsaved Tab
+   * @param componentData - New Tab
+   * @param saveDescriptionOnly
+   * @returns
+   */
+  public saveAPIRequest = async (
+    componentData: NewTab,
+    saveDescriptionOnly = false,
+  ) => {
+    const { folderId, folderName, collectionId, workspaceId } =
+      componentData.path;
+    const _collection = await this.readCollection(collectionId);
+    let userSource = {};
+    if (_collection?.activeSync && componentData?.source === "USER") {
+      userSource = {
+        currentBranch: _collection?.currentBranch,
+        source: "USER",
+      };
+    }
+    const _id = componentData.id;
+    let existingRequest;
+    if (!folderId) {
+      existingRequest = await this.readRequestOrFolderInCollection(
+        collectionId,
+        _id,
+      );
+    } else {
+      existingRequest = await this.readRequestInFolder(
+        collectionId,
+        folderId,
+        _id,
+      );
+    }
+    const bodyType =
+      componentData.property.request.state.dataset === RequestDataset.RAW
+        ? componentData.property.request.state.raw
+        : componentData.property.request.state.dataset;
+    let expectedRequest: RequestBody;
+    let expectedMetaData;
+    if (!saveDescriptionOnly) {
+      // Save overall api
+      expectedRequest = {
+        method: componentData.property.request.method,
+        url: componentData.property.request.url,
+        body: componentData.property.request.body,
+        headers: componentData.property.request.headers,
+        queryParams: componentData.property.request.queryParams,
+        auth: componentData.property.request.auth,
+        selectedRequestBodyType: setContentTypeHeader(bodyType),
+        selectedRequestAuthType: componentData.property.request.state?.auth,
+      };
+      expectedMetaData = {
+        id: _id,
+        name: componentData?.name,
+        description: componentData?.description,
+        type: ItemType.REQUEST,
+      };
+    } else {
+      // Save api description only
+      expectedRequest = {
```

Comment:
As discussed, can we move this mapping to a dedicated file or function? Would make the code more readable. A different `private` function in same file would also work. Take your call.